### ISSUE TITLE
Move Metadata Extraction Logic From `load_data` Method in `Project` Model to its Own Method

### ISSUE THREAD

**Issue by avrohomgottlieb** [AlexsLemonade/scpca-portal#635]
### Context

The `load_data` method inside the `Project` model accomplishes a variety of different tasks. Among those tasks includes generating readmes, combining metadata, and creating computed files.

Metadata is received in the form of three files: `project_metadata.csv`, `samples_metadata.csv` and within each sample (/ library) a `SCPCL<library_number>_metadata.json` file. Inside the `load_data` method, each of these files is parsed, the metadata is collected and combined at the sample level, and an accumulated `single_cell_metadata.tsv` file is created and outputted (or `spatial_metadata` if the project/sample is of the *spatial* modality).

### Problem or idea

The metadata extraction logic is too tightly coupled with the other tasks within the `load_data` method.

### Solution or next step

The solution is to bundle together all of the logic associated with metadata extraction within the`load_data` method, and move it to its own method within the `Project` model.

**Comment by davidsmejia** [comment]
This sounds right. I think we should limit this issue to handling these lines. And just extracting the nested logic into multiple functions.
- https://github.com/AlexsLemonade/scpca-portal/blob/dev/api/scpca_portal/models/project.py#L859-L992


I think we should just write another issue for handling moving the metadata processing into another function.
- https://github.com/AlexsLemonade/scpca-portal/blob/dev/api/scpca_portal/models/project.py#L807-L848

**Comment by avrohomgottlieb** [comment]
I agree with you on this. It makes sense to leave `get_samples` for another issue.

I think the following lines should also be included in the current issue:
- https://github.com/AlexsLemonade/scpca-portal/blob/dev/api/scpca_portal/models/project.py#L1116-L1138

**Comment by avrohomgottlieb** [comment]
@davidsmejia With single-cell data we receive the three metadata files listed above and output a `single_cell_metadata.tsv` file, but with spatial data we output both the combined `spatial_metadata.tsv` file and the input library json file (unmodified). Is that by design or a bug?

**Comment by davidsmejia** [comment]
This was addressed in #652 

Will move last question about spatial to another issue.
